usaspending_api/etl/management/commands/load_budget_authority.py

```diff
@@ -0,0 +1,165 @@
+
+
+def valid_quarter(raw):
+    result = int(raw)
+    if not 1 <= result <= 4:
+        raise ArgumentTypeError('Quarter should be 1-4')
+    return result
+
+
+class Command(BaseCommand):
+    """
+    Loads historical budget authority data from spreadsheets.
+
+    Expects the following files:
+
+        budget_authority.csv: Historical budget authority by FY
+        broker_rules_fr_entity.xlsx: fr_entity_codes by agency ID, MAC, subfunction code
+        quarterly/*.xls: quarter-by-quarter data for this fiscal year
+
+    Files in `quarterly` are only read if which quarter is specified
+    with the `--quarter` option.
+
+    By default, these files are expected in the
+    usaspending_api/data/budget_authority directory
+    (they are checked into the repository there).
+    A different directory can be searched instead
+    using the --directory option.  (This is done for
+    testing to test against smaller versions of the files.)
+    """
+
+    help = "Loads historical budget authority data from a CSV"
+    DEFAULT_DIRECTORY_PATH = os.path.join('usaspending_api', 'data',
+                                          'budget_authority')
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-q',
+            '--quarter',
+            type=valid_quarter,
+            help='Quarter to load from spreadsheets in data/budget_authority/quarterly',
+        )
+
+        parser.add_argument(
+            '-d',
+            '--directory',
+            default=self.DEFAULT_DIRECTORY_PATH,
+            help='Directory containing broker_rules_fr_entity.xlsx, budget_authority.csv, and quarterly/ directory',
+        )
+
+    def load_frec_map(self):
+
+        FREC_MAP_PATH = os.path.join(self.directory,
+                                     'broker_rules_fr_entity.xlsx')
+
+        workbook = open_workbook(FREC_MAP_PATH)
+        sheet = workbook.sheets()[1]
+        headers = [cell.value for cell in sheet.row(3)]
+
+        FrecMap.objects.all().delete()
+
+        instances = []
+        for i in range(4, sheet.nrows):
+            row = dict(zip(headers, (cell.value for cell in sheet.row(i))))
+            instance = FrecMap(
+                agency_identifier=row['AID'],
+                main_account_code=row['MAIN'],
+                treasury_appropriation_account_title=row['GWA_TAS NAME'],
+                sub_function_code=row['Sub Function Code'],
+                fr_entity_code=row['FR Entity Type'])
+            instances.append(instance)
+        FrecMap.objects.bulk_create(instances)
+
+    def load_quarterly_spreadsheets(self, quarter, results):
+        """Special procedure for getting quarterly update .xls files
+
+        These are downloaded from
+        MAX Information and Reports (Executive, Legislative, and Judicial Users)
+        https://max.omb.gov/maxportal/document/SF133/Budget/FY%202017%20-%20SF%20133%20Reports%20on%20Budget%20Execution%20and%20Budgetary%20Resources.html
```

Comment:
I don't think we have anything formal for it. We may need to add it to a README somewhere, or maybe on the wiki. But that shouldn't hold up the PR. It can happen independently.